javaslang-benchmark/src/test/java/javaslang/benchmark/JmhRunner.java

```diff
@@ -0,0 +1,66 @@
+package javaslang.benchmark;
+
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.results.RunResult;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.TimeValue;
+
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+
+public class JmhRunner {
+    private static final int WARMUP_ITERATIONS = 10;
+    private static final int MEASUREMENT_ITERATIONS = 40;
+
+    private static final int QUICK_WARMUP_ITERATIONS = 5;
+    private static final int QUICK_MEASUREMENT_ITERATIONS = 10;
+
+    public static void runAndReport(Class clazz) {
+        runAndReport(clazz, WARMUP_ITERATIONS, MEASUREMENT_ITERATIONS, Assertions.Disable);
+    }
+
+    public static void devRunAndReport(Class clazz) {
+        runAndReport(clazz, QUICK_WARMUP_ITERATIONS, QUICK_MEASUREMENT_ITERATIONS, Assertions.Enable);
+    }
+
+    private static void runAndReport(Class clazz, int warmupIterations, int measurementIterations, Assertions assertions) {
+        Collection<RunResult> results = run(clazz, warmupIterations, measurementIterations, assertions);
+        BenchmarkPerformanceReporter.of(results).print();
+    }
+
+    private static Collection<RunResult> run(Class clazz, int warmupIterations, int measurementIterations, Assertions assertions) {
+        final Options opts = new OptionsBuilder()
+                .include(clazz.getSimpleName())
+                .shouldDoGC(true)
+                .shouldFailOnError(true)
+                .mode(Mode.Throughput)
+                .timeUnit(TimeUnit.SECONDS)
+                .warmupTime(TimeValue.milliseconds(500))
+                .warmupIterations(warmupIterations)
+                .measurementTime(TimeValue.milliseconds(500))
+                .measurementIterations(measurementIterations)
+                .forks(1)
+                .jvmArgsAppend("-XX:+UseG1GC", "-Xss100m", "-Xms1g", "-Xmx1g", assertions.vmArg)
```

Comment:
I understand the rationale of changing NewGen to 100% of the heap.  I must admit, however, it makes me a bit uneasy, as it is an artificial setup.  However, I agree that in theory by removing all GCs we should get stable results.

I did a quick test. I run the `devRunAndReport` on `ListBenchmark` with container size = 10000 and a 4Gb heap. The first run with default NewGen size, and the second with NewGen size=100%.  No GC was triggered in the second run. 

However, to my surprise, there were still considerable swings from iteration to iteration, so not sure what to make of these results. 

First run, normal NewGen:
```
Target         Operation   Impl                         Params  Count            Score  ±     Error  
ListBenchmark  Iterate     slang_persistent              10000      6       32,704.013  ±     4.23%  
ListBenchmark  Iterate     scala_persistent              10000      6       35,181.237  ±     1.87%  
ListBenchmark  Iterate     scala_mutable                 10000      6       41,459.476  ±     2.74%  
ListBenchmark  Iterate     pcollections_persistent       10000      6       30,591.533  ±     7.57%  
ListBenchmark  Iterate     java_mutable_linked           10000      6       28,613.690  ±    12.88%  
ListBenchmark  Iterate     java_mutable                  10000      6       90,556.997  ±     6.98%  
ListBenchmark  Iterate     fjava_persistent              10000      6       35,779.948  ±     5.46%  
ListBenchmark  AddAll      slang_persistent              10000      6       26,124.064  ±    20.22%  
ListBenchmark  AddAll      scala_persistent              10000      6       20,810.554  ±     9.16%  
ListBenchmark  AddAll      scala_mutable                 10000      6       20,356.731  ±     1.28%  
ListBenchmark  AddAll      pcollections_persistent       10000      6       22,217.533  ±    36.81%  
ListBenchmark  AddAll      java_mutable_linked           10000      6       21,513.086  ±    20.40%  
ListBenchmark  AddAll      java_mutable                  10000      6       23,199.166  ±    52.08%  
ListBenchmark  AddAll      fjava_persistent              10000      6       14,222.033  ±     3.23%  
```

Second run, NewGen = 100%
```
Target         Operation   Impl                         Params  Count            Score  ±     Error  
ListBenchmark  Iterate     slang_persistent              10000      6       25,462.352  ±     1.42%  
ListBenchmark  Iterate     scala_persistent              10000      6       25,151.763  ±     2.27%  
ListBenchmark  Iterate     scala_mutable                 10000      6       44,893.597  ±     4.30%  
ListBenchmark  Iterate     pcollections_persistent       10000      6       25,520.437  ±     4.26%  
ListBenchmark  Iterate     java_mutable_linked           10000      6       30,922.490  ±     4.99%  
ListBenchmark  Iterate     java_mutable                  10000      6      112,805.879  ±     4.02%  
ListBenchmark  Iterate     fjava_persistent              10000      6       24,831.025  ±     9.02%  
ListBenchmark  AddAll      slang_persistent              10000      6       26,400.629  ±    17.79%  
ListBenchmark  AddAll      scala_persistent              10000      6       21,334.084  ±     4.55%  
ListBenchmark  AddAll      scala_mutable                 10000      6       19,364.354  ±     9.11%  
ListBenchmark  AddAll      pcollections_persistent       10000      6       23,310.641  ±    45.25%  
ListBenchmark  AddAll      java_mutable_linked           10000      6       22,698.641  ±    31.58%  
ListBenchmark  AddAll      java_mutable                  10000      6       23,148.883  ±    50.05%  
ListBenchmark  AddAll      fjava_persistent              10000      6       14,240.217  ±     2.97%  
```